base/util.go

```diff
@@ -1135,3 +1136,10 @@ func (ab *AtomicBool) Set(flag bool) {
 func (ab *AtomicBool) IsTrue() bool {
 	return atomic.LoadInt32(&ab.value) == 1
 }
+
+func Sha1HashString(string string, salt string) string {
+	h := sha1.New()
+	h.Write([]byte(string + salt))
```

Comment:
The sgcollect_info version has the salt as the first element, and then the data. We'll need to try and make sure these things hash the same way.

```suggestion
	h.Write([]byte(salt + str))
```